swiftui/ChartboostMediationDemo/Ads/Interstitial/InterstitialAdController.swift

```diff
@@ -44,47 +50,105 @@ class InterstitialAdController: NSObject, ObservableObject {
             return
         }
 
-        // Create an interstitial ad from the Chartboost Mediation ad provider.
-        guard let interstitialAd = chartboostMediation.interstitialAdProvider(with: self, andPlacementName: placementName) else {
-            // It could fail if the placement name is invalid.
-            print("[Error] failed to create interstitial advertisement for placement '\(placementName)'")
-            return
-        }
-        self.interstitialAd = interstitialAd
-
         // Notify the demo UI
         activityState = .running
 
-        // Associate any provided keywords with the advertisement before it is loaded.
-        interstitialAd.keywords = keywords
+        if fullscreenAPI {
+            // loadFullscreenAd expects keywords to be `[String : String]` instead of `HeliumKeywords?`.
+            let keywords = keywords?.dictionary ?? [:]
+            let request = ChartboostMediationAdLoadRequest(placement: placementName, keywords: keywords)
+            // Load the fullscreen ad, which will make a request to the network. Upon completion, a
+            // ChartboostMediationFullscreenAdLoadResult will be passed to the completion block.
+            chartboostMediation.loadFullscreenAd(with: request) { [weak self] result in
+                guard let self = self else { return }
+                if let ad = result.ad {
+                    ad.delegate = self
+                    self.fullscreenAd = result.ad
 
-        // Load the interstitial ad, which will make a request to the network. Upon completion, the
-        // delegate method `heliumInterstitialAd(withPlacementName:didLoadWithError:)` will be called.
-        interstitialAd.load()
+                    log(action: "load", placementName: placementName, error: result.error)
+                    // Notify the demo UI
+                    activityState = .idle
+                } else {
+                    fullscreenAd = nil
+                    self.log(action: "Load error", placementName: placementName, error: result.error)
+
+                    // `.failed` requires a non-optional error type
+                    let error = result.error ?? NSError(domain: "com.chartboost.mediation.demo", code: 0)
```

Comment:
I'd like to use something that makes more sense than `code: 0` here but I didn't see any NSError Codes that seemed applicable. Does anyone know a generic NSError Code that could work here?  All the 'unknown' errors I saw were domain-specific like `NSFileReadUnknownError`

Reply:
Why don't you make the error parameter optional in ActivityState.failed ? It will be cleaner on the point of use.